### PR TITLE
Benchmarks for a lot of functions

### PR DESCRIPTION
**Pull request by Coding-Doctors**
Just some initial ones for now, having issues with my editor ATM

### CODE REVIEW

File: benches/benchmarks.rs [hunk 1]
```diff
@@ -1,13 +1,31 @@
 #![feature(test)]
 
-extern crate serenity_utils;
-extern crate test;
+mod benchs {
```
Comment (zeyla) [inline benches/benchmarks.rs]: What's the reason for putting everything in a module?

File: benches/benchmarks.rs [hunk 2]
```diff
@@ -1,13 +1,31 @@
...
+    extern crate serenity_utils as U;
```
Comment (zeyla) [inline benches/benchmarks.rs]: Let's alias this as `utils`

File: benches/benchmarks.rs [hunk 3]
```diff
@@ -1,13 +1,31 @@
...
-use test::{Bencher, black_box};
+    use self::test::{Bencher, black_box};
 
-#[bench]
-fn bench_parse_emoji(b: &mut Bencher) {
-    b.iter(|| {
-        serenity_utils::parse_emoji("<:smugAnimeFace:302516740095606785>");
-    });
+    #[bench]
+    fn pharse_emoji(b: &mut Bencher) {
```
Comment (zeyla) [inline benches/benchmarks.rs]: `s/pharse/parse`

File: benches/benchmarks.rs [hunk 4]
```diff
@@ -1,13 +1,31 @@
...
+        b.iter(|| {
+            U::parse_emoji("<:smugAnimeFace:302516740095606785>");
+        });
+    }
+
+    #[bench]
+    fn pharse_quotes(b: &mut Bencher) {
```
Comment (zeyla) [inline benches/benchmarks.rs]: `s/pharse/parse`

File: benches/benchmarks.rs [hunk 5]
```diff
@@ -1,13 +1,31 @@
...
+            U::parse_quotes(r#""this is the first" "this is the second""#);
+        })
+    }
+
+    #[cfg(feature = "cache")]
+    #[bench]
+    fn with_cache(b: &mut Bencher) {
+        b.iter(|| {
+            U::with_cache(|cache| cache.user.id);
+
```
Comment (zeyla) [inline benches/benchmarks.rs]: There's an extra newline here

File: benches/benchmarks.rs (hunk 1 above)
Comment (Coding-Doctors) [reply benches/benchmarks.rs]: Looks prettier
screenshot_2017-11-21_19-18-13

File: benches/benchmarks.rs (hunk 2 above)
Comment (Coding-Doctors) [reply benches/benchmarks.rs]: Sure, did that for convience, makes sense to make it readable first.

Comment (Coding-Doctors) [reply benches/benchmarks.rs]: Done!

**Comment by Coding-Doctors** [comment]
Feel free to review this, I'm finished with what I orginally set out to change in this pr

**Comment by zeyla** [comment]
Thanks!

**Pull request merged**